lib/ex_aws/request.ex

```diff
@@ -84,9 +86,11 @@ defmodule ExAws.Request do
     {:attempt, attempt + 1}
   end
 
-  # TODO: make exponential
-  # TODO: add jitter
   def backoff(attempt) do
-    :timer.sleep(attempt * 1000)
+    (@base_backoff_in_ms * :math.pow(2, attempt))
+    |> min(@max_backoff_in_ms)
+    |> trunc
+    |> :rand.uniform
```

Comment:
Hm interesting, I do actually remember reading that article now when it came out. I suppose the reason I was leaning toward "equal" jitter was because the scenario I'm using fallback for is to compensate for network irregularities instead of throughput. The general assumption is that the AWS service being hit is more than capable of handling whatever load our system wants to throw at them. Thus if we're hitting issues, it's some kind of intermittent network issue, and with those it is important to wait out the issue (within bounds).

That said, I have no empirical evidence that equal jitter is better at avoiding network issues. I suppose there's also an argument to be made in favor of full jitter here in such cases, because by minimizing the probability that a group of retries all happen at once, we minimize local network load from the box.

Long story short, I think your approach here with full jitter is likely correct.